File: go/muse/util/random_factory/random_factory.go
package random_factory

import (
	"math/rand/v2"
)

func GenIntN(min, max int) int {
	return min + rand.N(max-min+1)
}

func GenInt() int {
	return GenIntN(1, 2_147_483_647)
}

func GenEven() int {
	return GenInt() >> 1 << 1
}

func GenOdd() int {
	return GenInt() | 1
}

File: archive/php/muse/util/RandomFactory.php
<?php

declare(strict_types=1);

namespace muse\util\RandomFactory;

function genIntN(int $min, int $max): int
{
    return random_int($min, $max);
}

function genInt(): int
{
    return genIntN(1, 2_147_483_647);
}

function genEven(): int
{
    return genInt() >> 1 << 1;
}

function genOdd(): int
{
    return genInt() | 1;
}
